src/elli_http.erl

```diff
@@ -588,21 +588,33 @@ do_check_max_size_x2(_, _, _, _) -> ok.
       Socket    :: elli_tcp:socket() | undefined,
       Callback  :: elli_handler:callback(),
       Req       :: elli:req().
-mk_req(Method, RawPath, Headers, Body, V, Socket, {Mod, Args} = Callback) ->
-    case parse_path(RawPath) of
-        {ok, {Path, URL, URLArgs}} ->
-            #req{method   = Method, path     = URL,    args    = URLArgs,
+mk_req(Method, PathTuple, Headers, Body, V, Socket, {Mod, Args} = Callback) ->
+    case parse_path(PathTuple) of
+        {ok, {SchemeHostPort, {Path, URL, URLArgs}}} ->
+            {Scheme, Host, Port} = case SchemeHostPort of
+                undefined ->
+                    list_to_tuple(lists:duplicate(3, undefined));
```

Comment:
Can't `parse_path` just always return the 3-tuple so none of this has to be done. And even if for some reason we can't have `parse_path` handle that this should just be `{undefined, undefined, undefined}`.

Reply:
Yep.

Reply:
Oh funny `-compile(inline_list_funcs).` does not inline `lists:duplicate/2`